src/org/epics/pvdata/copy/arrayPlugin/ArrayFilter.java

```diff
@@ -0,0 +1,149 @@
+
+/*
+ * The License for this software can be found in the file LICENSE that is included with the distribution.
+ */
+package org.epics.pvdata.copy.arrayPlugin;
+
+import org.epics.pvdata.copy.PVFilter;
+import org.epics.pvdata.factory.ConvertFactory;
+import org.epics.pvdata.misc.BitSet;
+import org.epics.pvdata.pv.Convert;
+import org.epics.pvdata.pv.PVField;
+import org.epics.pvdata.pv.PVScalarArray;
+import org.epics.pvdata.pv.Type;
+/**
+ * A filter that gets a sub array from a PVScalarArray.
+ * @author mrk
+ * @since 2017.02.23
+ */
+public class  ArrayFilter implements PVFilter{
+    static final Convert convert = ConvertFactory.getConvert();
+    PVScalarArray masterArray;
+    int start;
+    int increment;
+    int end;
+
+    public static ArrayFilter create(String requestValue,PVField master)
+    {
+        Type type = master.getField().getType();
+        if(type!=Type.scalarArray) {
+            return null;
+        }
+        int start =0;
+        int increment =1;
+        int end = -1;
+        String[] values = requestValue.split(":");
+        int num = values.length;
+        if(num==1) {
+            String value = values[0];
+            try{
```

Comment:
This `try-catch` block is used multiple times. Please consider extract it into a method.